File: js/monde/Environnement.js
(function(){

if(!window.creatura)
	window.creatura={};

/**
 * Représente les données envrionnementales d'une zone ou celles affectant une créature
 */
creatura.Environnement = function(temperature, luminosite, couleur){
	this.temperature = temperature;
	this.luminosite = luminosite;
	this.couleur = couleur;
	
	/**
	 * Renvoie la couleur de l'environnement
	 */
	this.getCouleur = function(){
		return this.couleur;
	};
	
	/**
	 * Renvoie la température de l'environnement
	 */
	this.getTemperature = function(){
		return this.temperature;
	};
	
	/**
	 * Renvoie la luminosité de l'envrionnement 
	 */	
	this.getLuminosite = function(){
		return this.luminosite;
	};
};
	
})();

File: src/net/feerie/creatura/shared/Environnement.java
package net.feerie.creatura.shared;

/**
 * Représente les données envrionnementales d'une zone ou celles affectant une
 * créature
 */
public class Environnement
{
	private int temperature;
	private double luminosite;
	private String couleur;
	
	/**
	 * Constructeur
	 * 
	 * @param temperature la température (en degrés celsius)
	 * @param luminosite la luminosité (de 0.0f à 1.0f)
	 * @param couleur le code couleur de l'environnement
	 */
	public Environnement(int temperature, double luminosite, String couleur)
	{
		super();
		this.temperature = temperature;
		this.luminosite = luminosite;
		this.couleur = couleur;
	}
	
	/**
	 * @return la température de l'environnement
	 */
	public int getTemperature()
	{
		return temperature;
	}
	
	/**
	 * @return la luminosité de l'environnement
	 */
	public double getLuminosite()
	{
		return luminosite;
	}
	
	/**
	 * @return la couleur de l'environnement
	 */
	public String getCouleur()
	{
		return couleur;
	}
}
